packages/reactivity/src/ref.ts

```diff
@@ -85,10 +110,50 @@ declare const ShallowRefMarker: unique symbol
 
 export type ShallowRef<T = any> = Ref<T> & { [ShallowRefMarker]?: true }
 
+/**
+ * Creates a ref that tracks mutations on its own `value` but doesn't make its
+ * value reactive.
+ *
+ * @example
+ * ```js
+ * const foo = shallowRef({})
+ * // mutating the ref's value is reactive
+ * foo.value = {}
+ * // but the value will not be converted.
+ * isReactive(foo.value) // false
+ * ```
+ *
+ * @see {@link https://v3.vuejs.org/guide/reactivity-fundamentals.html#creating-standalone-reactive-values-as-refs}
+ *
+ * @param value The "inner value" for the shallow ref.
+ */
 export function shallowRef<T extends object>(
   value: T
 ): T extends Ref ? T : ShallowRef<T>
+/**
+ * Creates a ref that tracks mutations on its own `value` but doesn't make its
+ * value reactive.
+ *
+ * @example
+ * ```js
+ * const foo = shallowRef({})
+ * // mutating the ref's value is reactive
+ * foo.value = {}
+ * // but the value will not be converted.
+ * isReactive(foo.value) // false
+ * ```
+ *
+ * @see {@link https://v3.vuejs.org/guide/reactivity-fundamentals.html#creating-standalone-reactive-values-as-refs}
+ *
+ * @param value The "inner value" for the shallow ref.
+ */
 export function shallowRef<T>(value: T): ShallowRef<T>
+/**
+ * Creates a ref that tracks mutations on its own `value` but doesn't make its
+ * value reactive.
+ *
+ * @see {@link https://v3.vuejs.org/guide/reactivity-fundamentals.html#creating-standalone-reactive-values-as-refs}
+ */
```

Comment:
We can probably just use one doc block for this one, no?

Reply:
Not so sure about this. Yeah, the repetition kinda sucks. Back when I wrote this, the documentation would get picked up by intelliJ only for the overloads of the method that actually have the doc block on them. That's maybe just an intelliJ inconvenience and it definitely makes more sense to reduce boilerplate / repetition in the source code.

Reply:
That sounds more like an IntelliJ bug. I think it would be better to report it, they seem to be very active in fixing bugs so it would be better not to let this stagnate

Reply:
Seems intelliJ has already fixed this as I couldn't reproduce this weird behaivour again.